apps/site/pages/en/blog/announcements/making-nodejs-downloads-reliable.md

```diff
@@ -0,0 +1,221 @@
+---
+date: '2024-12-28T12:00:00.000Z'
+category: announcements
+title: Making Node.js Downloads Reliable
+layout: blog-post
+author: flakey5
+---
+
+Last year, we shared [the details behind Node.js' brand new website](https://nodejs.org/en/blog/announcements/diving-into-the-nodejs-website-redesign).
+Today we're back, talking about the new infrastructure serving Node.js' release assets.
+
+<!-- I don't really like this paragraph -->
+
+This blog post goes into what Node.js' web infrastructure looks like, its history, and where it stands today.
+We're also going to be covering what exactly we had in mind and were prioritizing with this infrastructure overhaul.
+
+## Some Brief History
+
+At the start of the project in 2009, Node.js release assets (binaries and documentation) were stored on a publicly accessible S3 bucket.
+This was later changed around May of 2010 to a VPS, which also hosted the Node.js website.
+
+When the io.js fork happened in 2014, they also had used a VPS to host the website and release assets.
+After the Node.js and io.js merge in 2015, io.js' VPS (which will be referred to as the origin server from hereon) was repurposed to host both io.js and Node.js releases, along with the Node.js website.
+
+A backup server was also created, which acted almost like an exact copy of the origin server.
+It served two main purposes:
+
+1. Serve any traffic that the origin server couldn't handle
+2. Serve as a backup for the binaries and documentation in case if something went wrong with the origin server.
+
+The entire architecture looked like this:
+
+![A diagram of the old infrastructure. Cloudflare is used as a cache and load balancer. If the origin server is healthy, traffic is sent to it. Otherwise, the backup server receives it.](/static/images/blog/announcements/old-release-asset-infra.png)
+
+### How Node.js Releases are Released
+
+When a mainline release is ready to be released, the releaser triggers a release CI job.
+These jobs build the Node.js binaries for all supported platforms and generate the documentation files.
+
+Upon a release CI job completing successfully, its assets are uploaded to the origin server in a staging directory.
+When all of the release CI jobs finish, the releaser then manually "promotes" the assets to the production directory, making the release publicly accessible.
+
+For nightly or test releases, the promotion happens automatically.
+
+_Refs: [Release Overview](https://github.com/nodejs/build/blob/main/doc/release-overview.md), [Node.js release process](https://github.com/nodejs/node/blob/main/doc/contributing/releases.md)_.
+
+## Growing Pains
+
+Nowadays, the nodejs.org domain sees over 3 billion requests and 2+ petabytes of traffic per month, with the majority of that going towards release assets.
```

Comment:
@ovflowd could you confirm these numbers?

Reply:
2.41b requests / 3.18Pb traffic